x/gov/abci.go

```diff
@@ -72,6 +72,103 @@ func EndBlocker(ctx sdk.Context, keeper *keeper.Keeper) error {
 		return err
 	}
 
+	// fetch proposals that are due to be checked for quorum
+	rng = collections.NewPrefixUntilPairRange[time.Time, uint64](ctx.BlockTime())
+	err = keeper.QuorumCheckQueue.Walk(ctx, rng, func(key collections.Pair[time.Time, uint64], quorumCheckEntry v1.QuorumCheckQueueEntry) (bool, error) {
+		proposal, err := keeper.Proposals.Get(ctx, key.K2())
+		if err != nil {
+			return false, err
+		}
+
+		var tagValue, logMsg string
+
+		params, err := keeper.Params.Get(ctx)
+		if err != nil {
+			return false, err
+		}
+
+		// remove from queue
+		err = keeper.QuorumCheckQueue.Remove(ctx, key)
+		if err != nil {
+			return false, err
+		}
+
+		// check if proposal passed quorum
+		quorum, err := keeper.HasReachedQuorum(ctx, proposal)
+		if err != nil {
+			return false, err
+		}
+
+		logMsg = "proposal did not pass quorum after timeout, but was removed from quorum check queue"
+		tagValue = types.AttributeValueProposalQuorumNotMet
+
+		if quorum {
+			logMsg = "proposal passed quorum before timeout, vote period was not extended"
+			tagValue = types.AttributeValueProposalQuorumMet
+			if len(quorumCheckEntry.QuorumCheckTimestamps) == 0 {
```

Comment:
IIRC this was `quorumCheckCount > 0`, so I would have expect to see `len(quorumCheckEntry.QuorumCheckTimestamps) > 0`  right ?

Reply:
yes! lol sorry, my bad. Thanks for catching it!

Reply:
We really need tests :)